plugins/modules/ec2_vpc_vgw.py

```diff
@@ -176,7 +190,7 @@ def found(response_code, catch_extra_error_codes=None):
         return False
 
 
-def get_vgw_info(vgws):
+def get_vgw_info(vgws: List[Dict[str, Any]]) -> Optional[Dict[str, Any]]:
```

Comment:
```suggestion
def get_vgw_info(vgw: Dict[str, Any]) -> Dict[str, Any]:
```
The function should be renamed to `format_vpn_gateway()` and accept a `dict` as input, not a list of dicts. We should not loop on vgws but threat the single element.
We should check whether the input is not an empty dict to deal with the specific case of `check_mode`

Reply:
fixed